libs/api/domains/identity/src/lib/models/identity.model.ts

```diff
@@ -0,0 +1,68 @@
+import {
+  Field,
+  Extensions,
+  MiddlewareContext,
+  ObjectType,
+  InterfaceType,
+  ID,
+  registerEnumType,
+} from '@nestjs/graphql'
+
+import { NationalRegistryUser } from '@island.is/api/domains/national-registry'
+
+import { IdentityType } from '../identity.type'
+import { Address } from './address.model'
+
+registerEnumType(IdentityType, { name: 'IdentityType' })
+
+@InterfaceType({
+  resolveType(identity: Identity) {
+    switch (identity.type) {
+      case IdentityType.Person:
+        return IdentityPerson
+      case IdentityType.Company:
+        return IdentityCompany
+      default:
+        return null
+    }
+  },
+})
+export class Identity {
+  @Field(() => ID)
+  nationalId!: string
+
+  @Field(() => String)
+  name!: string
+
+  @Field(() => Address, { nullable: true })
+  address?: Address
+
+  @Field(() => IdentityType)
+  type!: IdentityType
+}
+
+const validateIf = ({ context, source }: MiddlewareContext): boolean => {
+  return (context as any).req.user.nationalId !== source.nationalId
```

Comment:
Would be nice to create our own graphql context type. I think we have somewhere in auth-nest-tools a type of `request` with our user/auth attributes.

Should conditionally read `user` as it can be null?

Reply:
I assumed this wouldn't get called since its resolvers are protected with auth and thus at this stage, the req.user will always be set.


But I agree with our own graphql context type. Will add it.